From f539d96ef4a34ad23d33c552d1770ab429b0220e Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 14 Jun 2023 00:24:36 -0700
Subject: [PATCH] tracefs-perf: Add missing headers for syscall() and SYS_*
 defines

These headers are needed to provide prototype definitions for functions
e.g. close(), syscall(), getpagesize(), getpid()

The issue is observed with clang-16+ compiler on musl systems

Upstream-Status: Submitted [https://lore.kernel.org/linux-trace-devel/20230614072759.316154-1-raj.khem@gmail.com/T/#u]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/tracefs-perf.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/tracefs-perf.c b/src/tracefs-perf.c
index 96d12cd..62c1508 100644
--- a/src/tracefs-perf.c
+++ b/src/tracefs-perf.c
@@ -1,4 +1,5 @@
-#include <asm/unistd.h>
+#include <unistd.h>
+#include <sys/syscall.h>
 #include <sys/mman.h>
 #include <signal.h>
 #include <linux/perf_event.h>
-- 
2.41.0

